owncloud/ocis-charts
# move postprocessing restarts into one cronjob

## Description
Replaces the two postprocessing restart maintenance cronjobs with one storageusers postprocessing restart maintenance cronjob.

## Related Issue
- https://github.com/owncloud/ocis/pull/8782 / https://github.com/owncloud/ocis/pull/8811 , https://github.com/owncloud/enterprise/issues/6533


## Motivation and Context
Also restart postprocessing for files that are not in the postprocessing store, but only on disk.

## How Has This Been Tested?
- minikube with ocis-nats deployment example:

1. make all files stuck in postprocessing: `kubectl exec -n ocis-nats deployments/nats-box -- sh -c 'nats consumer rm -f main-queue antivirus; nats consumer rm -f main-queue audit; nats consumer rm -f main-queue clientlog; nats consumer rm -f main-queue dcfs; nats consumer rm -f main-queue evhistory; nats consumer rm -f main-queue frontend; nats consumer rm -f main-queue notifications; nats consumer rm -f main-queue policies; nats consumer rm -f main-queue postprocessing; nats consumer rm -f main-queue search; nats consumer rm -f main-queue storage-users; nats consumer rm -f main-queue userlog'`
2. upload files and confirm they are in postprocessing (`kubectl -n ocis exec deployment/storageusers -it -- ocis storage-users uploads list`)
3. restart all ocis pods to recreate nats consumers: `k rollout restart deploy -n ocis`
4. enable cronjob:
```diff
diff --git a/deployments/ocis-nats/helmfile.yaml b/deployments/ocis-nats/helmfile.yaml
index 61461fa..2639a17 100644
--- a/deployments/ocis-nats/helmfile.yaml
+++ b/deployments/ocis-nats/helmfile.yaml
@@ -73,6 +73,9 @@ releases:
     namespace: ocis
     chart: ../../charts/ocis
     values:
+      - image:
+          repository: wkloucek/ocis
+          tag: release-5.0.1-8d1ba0244c3203c9bd2ac05b683d01afe885fc8d
       - externalDomain: ocis.kube.owncloud.test
       - ingress:
           enabled: true
@@ -130,6 +133,10 @@ releases:
           storageusers:
             persistence:
               enabled: true
+            maintenance:
+              restartPostprocessing:
+                enabled: true
+                schedule: "*/5 * * * *"
 
           thumbnails:
             persistence:
```
5. wait for the cronjob to run. see that it actually restarted the postprocessing for the files you had stuck in postprocessing
6. verify that the files are no longer in postprocessing via the web ui

## Screenshots (if appropriate):

## Types of changes

## Checklist:
- [x] Documentation generated (`make docs`) and committed
- [ ] Documentation PR created: <link>
